crates/uv-git/src/sha.rs

```diff
@@ -30,9 +33,77 @@ impl std::fmt::Display for GitSha {
 }
 
 impl FromStr for GitSha {
-    type Err = git2::Error;
+    type Err = OidParseError;
 
     fn from_str(value: &str) -> Result<Self, Self::Err> {
-        Ok(Self(git2::Oid::from_str(value)?))
+        Ok(Self(GitOid::from_str(value)?))
+    }
+}
+
+/// Unique identity of any Git object (commit, tree, blob, tag).
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct GitOid {
+    len: usize,
+    bytes: [u8; 40],
+}
+
+impl GitOid {
+    /// Return the string representation of an object ID.
+    pub(crate) fn as_str(&self) -> &str {
+        str::from_utf8(&self.bytes[..self.len]).unwrap()
+    }
+}
+
+#[derive(Debug, Error, PartialEq)]
+pub enum OidParseError {
+    #[error("Object ID can be at most 40 hex characters")]
+    TooLong,
+    #[error("Object ID cannot be parsed from empty string")]
+    Empty,
+}
+
+impl FromStr for GitOid {
+    type Err = OidParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.is_empty() {
+            return Err(OidParseError::Empty);
+        }
+
+        if s.len() > 40 {
+            return Err(OidParseError::TooLong);
+        }
+
+        let mut out = [0; 40];
+        out[..s.len()].copy_from_slice(s.as_bytes());
```

Comment:
Do we need to validate that this a valid hex?

Reply:
I don't think so, git will end up failing if we try to identify a commit using a malformed/broken hash anyways.